script.js

```diff
@@ -1,55 +1,69 @@
-  
-}
-
-Animal.prototype.putInTheDocument = function(){
-    var petsTable = document.getElementById("petsTable");
-    var petTR = document.createElement("tr");
-  
-    var petNameTD = document.createElement("td");
-    petNameTD.textContent = this.name;
-    petTR.appendChild(petNameTD);
-  
-    var petLegsTD = document.createElement("td");
-    petLegsTD.textContent = this.legs;
-    petTR.appendChild(petLegsTD);
-  
-    var petActionTD = document.createElement("td");
-    var petActionTDButton = document.createElement("button");
-    petActionTDButton.textContent = this.actionText;
-    petActionTD.appendChild(petActionTDButton);
-    petTR.appendChild(petActionTD);
+class Animal{
+    constructor(name){
+          this.name =name;
+          
+      }
+    action(){
+       
+          document.getElementById(this.actionSoundName).play();
+  
+      }
+  
+    goImage(){
+          let mainImage =document.querySelector('img');
+          mainImage.src=this.showImage;
+      }
+    putInTheDocument(){
+      var petsTable = document.getElementById("petsTable");
+      var petTR = document.createElement("tr");
+    
+      var petNameTD = document.createElement("td");
+      petNameTD.textContent = this.name;
+      petTR.appendChild(petNameTD);
+    
+      var petLegsTD = document.createElement("td"); 
+      petLegsTD.textContent = this.legs;
+      petTR.appendChild(petLegsTD);
     
-    petActionTDButton.onclick = this.action.bind(this);
-    petsTable.querySelector("tbody").appendChild(petTR)
-}
-
-
-function Cat(name){
-    Animal.call(this, name);
-    this.legs = 4;
-    this.actionText = "Meoow"
-    this.actionSoundName = "meow"
-}
-
-Cat.prototype = Animal.prototype;
-
-function Monkey(name){
-    Animal.call(this, name);
-    this.legs = 2;
-    this.actionText = "Scream";
-    this.actionSoundName = "scream";
-}
-
-Monkey.prototype = Animal.prototype;
-
-var Mila = new Cat("Mila");
-Mila.putInTheDocument();
-
-var Charlie = new Monkey("Charlie");
-Charlie.putInTheDocument();
+      var petActionTD = document.createElement("td");
+      var petActionTDButton = document.createElement("button");
+      petActionTDButton.textContent = this.actionText;
+      petActionTD.appendChild(petActionTDButton);
+      petTR.appendChild(petActionTD);
+      
+      petActionTDButton.onclick = this.action.bind(this);
+      petsTable.querySelector("tbody").appendChild(petTR);
+  
+      petTR.addEventListener('click',this.goImage.bind(this));
+  
+  }
+  }
+  
+  class Cat extends Animal{
+      constructor(name,legs,actionText,showImage,actionSoundName){
+          super(name);
+          this.legs=4;
+          this.actionSoundName='meow';
+          this.actionText='Meoow';
+          this.showImage='https://image.shutterstock.com/image-photo/18-july-2014-turkey-istanbul-260nw-526951102.jpg'
+      }
+  }
+  
+  
+  
+  class Monkey extends Animal{
+      constructor(name,legs,actionText,showImage,actionSoundName){
```

Comment:
Burada constructorın bütün bunları argüman olarak almasına gerek yok aslında. Eğer new Cat veya new Charlie ile yeni bir class verirken örneğin new Cat(Mila, 4) diyorsan içeride constructora(name,legs) verebilirsin ancak sen zaten classların içinde statik olarak vermişsin yani this.legs = 2 demişsin. Demek ki bu dışarıdan bir değişkenden almıyor leg sayısını. Öyleyse bir daha constructora argüman olarak geçmene gerek kalmıyor.